xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -2257,8 +2257,8 @@ std::string CAMLCodec::GetStereoMode()
     default:                                  stereo_mode = m_hints.stereo_mode; break;
   }
 
-  if(m_processInfo.GetVideoSettings().m_StereoInvert)
-    stereo_mode = RenderManager::GetStereoModeInvert(stereo_mode);
+  if (m_processInfo.GetVideoSettings().m_StereoInvert)
+    ;//stereo_mode = RenderManager::GetStereoModeInvert(stereo_mode);
```

Comment:
@FernetMenta I have no idea how to implement this one

Reply:
@peak3d My implementation:
```
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -2248,18 +2248,26 @@ bool CAMLCodec::SetVideo3dMode(const int mode3d)

 std::string CAMLCodec::GetStereoMode()
 {
-  std::string  stereo_mode;
+  std::string stereoMode;

   switch(m_processInfo.GetVideoSettings().m_StereoMode)
   {
-    case RENDER_STEREO_MODE_SPLIT_VERTICAL:   stereo_mode = "left_right"; break;
-    case RENDER_STEREO_MODE_SPLIT_HORIZONTAL: stereo_mode = "top_bottom"; break;
-    default:                                  stereo_mode = m_hints.stereo_mode; break;
+    case RENDER_STEREO_MODE_SPLIT_VERTICAL:
+      stereoMode = "left_right";
+      if (m_processInfo.GetVideoSettings().m_StereoInvert)
+        stereoMode = "right_left";
+      break;
+    case RENDER_STEREO_MODE_SPLIT_HORIZONTAL:
+      stereoMode = "top_bottom";
+      if (m_processInfo.GetVideoSettings().m_StereoInvert)
+        stereoMode = "bottom_top";
+      break;
+    default:
+      stereoMode = m_hints.stereo_mode;
+      break;
   }

-  if(m_processInfo.GetVideoSettings().m_StereoInvert)
-    stereo_mode = RenderManager::GetStereoModeInvert(stereo_mode);
-  return stereo_mode;
+  return stereoMode;
 }
```

Reply:
thanx @codesnake !

Reply:
@codesnake your code is not correct. A stereo mode set by decoder is not supposed to get overridden. see VPV

Reply:
@FernetMenta does this mean if m_hints.stereo_mode is valid is has to be uset with priority?

Reply:
no, check vpv
- if decoder sets stereo mode, use this
- if not set by decoder use udes setting
- if none of the above applies, se hints